src/main/docker/Dockerfile.jvm

```diff
@@ -1,96 +1,51 @@
-#   be greater than 4096MB. The value of this variable is expressed in MB (example: "4096")
-# - JAVA_DIAGNOSTICS: Set this to get some diagnostics information to standard output
-#   when things are happening. This option, if set to true, will set
-#  `-XX:+UnlockDiagnosticVMOptions`. Disabled by default (example: "true").
-# - JAVA_DEBUG: If set remote debugging will be switched on. Disabled by default (example:
-#    true").
-# - JAVA_DEBUG_PORT: Port used for remote debugging. Defaults to 5005 (example: "8787").
-# - CONTAINER_CORE_LIMIT: A calculated core limit as described in
-#   https://www.kernel.org/doc/Documentation/scheduler/sched-bwc.txt. (example: "2")
-# - CONTAINER_MAX_MEMORY: Memory limit given to the container (example: "1024").
-# - GC_MIN_HEAP_FREE_RATIO: Minimum percentage of heap free after GC to avoid expansion.
-#   (example: "20")
-# - GC_MAX_HEAP_FREE_RATIO: Maximum percentage of heap free after GC to avoid shrinking.
-#   (example: "40")
-# - GC_TIME_RATIO: Specifies the ratio of the time spent outside the garbage collection.
-#   (example: "4")
-# - GC_ADAPTIVE_SIZE_POLICY_WEIGHT: The weighting given to the current GC time versus
-#   previous GC times. (example: "90")
-# - GC_METASPACE_SIZE: The initial metaspace size. (example: "20")
-# - GC_MAX_METASPACE_SIZE: The maximum metaspace size. (example: "100")
-# - GC_CONTAINER_OPTIONS: Specify Java GC to use. The value of this variable should
-#   contain the necessary JRE command-line options to specify the required GC, which
-#   will override the default of `-XX:+UseParallelGC` (example: -XX:+UseG1GC).
-# - HTTPS_PROXY: The location of the https proxy. (example: "myuser@127.0.0.1:8181")
-# - HTTP_PROXY: The location of the http proxy. (example: "myuser@127.0.0.1:8181")
-# - NO_PROXY: A comma separated lists of hosts, IP addresses or domains that can be
-#   accessed directly. (example: "foo.example.com,bar.example.com")
-#
-###
+# ---- Stage 1: JDK Fetch and Setup ----
+FROM registry.access.redhat.com/ubi9/ubi AS data-api-jdk-builder
 
-# see https://catalog.redhat.com/software/containers/ubi8/openjdk-21-runtime/653fd184292263c0a2f14d69?gs&q=openjdk%2021%20ubi
-FROM registry.access.redhat.com/ubi8/openjdk-21-runtime:1.21-2
+# Latest JDK 21
+ARG JDK_VERSION=21.0.7_6
+ARG JDK_BUILD=jdk-21.0.7%2B6
+ARG JDK_DIST=OpenJDK21U-jdk_x64_linux_hotspot_21.0.7_6.tar.gz
 
-ENV LANGUAGE='en_US:en'
+RUN dnf install -y gzip tar findutils && \
+    curl -L -o /tmp/jdk.tar.gz https://github.com/adoptium/temurin21-binaries/releases/download/${JDK_BUILD}/${JDK_DIST} && \
+    mkdir -p /opt/java && \
+    tar -xzf /tmp/jdk.tar.gz -C /opt/java && \
+    mv /opt/java/jdk-* /opt/java/openjdk && \
+    rm /tmp/jdk.tar.gz && \
+    dnf clean all
 
+# ---- Stage 2: Final Runtime Image ----
+FROM registry.access.redhat.com/ubi9/ubi-minimal as data-api-final
+LABEL maintainer="DataStax, Inc <info@datastax.com>"
+LABEL name="data-api-container"
+LABEL vendor="DataStax, Inc"
+LABEL release="1.0.28"
```

Comment:
This needs to be documented so we'll update it as part of Data API release process.
(unless we can somehow pass it as ENV variable?)